src/features/collections/CollectionDetail.tsx

```diff
@@ -977,3 +1001,67 @@ const TextFilterControls = ({
     />
   );
 };
+
+const BooleanFilterControls = ({
+  column,
+  filter,
+  collectionId,
+  context,
+}: FilterControlProps & {
+  filter: { type: 'bool' };
+}) => {
+  const dispatch = useAppDispatch();
+  const [selectValue, setSelectValue] = useState<string>(() => {
+    if (filter.value === true || filter.value === 1) {
+      return 'true';
+    } else if (filter.value === false || filter.value === 0) {
+      return 'false';
+    } else {
+      return 'any';
+    }
+  });
+
+  const handleChange = (event: SelectChangeEvent) => {
+    setSelectValue(event.target.value as string);
+  };
+
+  // In order not to create a dependency loop when filter changes within the below effect,
+  // use a filter ref
+  const filterRef = useRef(filter);
+  filterRef.current = filter;
+
+  useEffect(() => {
+    let value;
+    if (selectValue === 'true') {
+      value = 1;
+    } else if (selectValue === 'false') {
+      value = 0;
+    }
+    dispatch(
+      setFilter([
+        collectionId,
+        context,
+        column,
+        { ...filterRef.current, value },
+      ])
+    );
+  }, [selectValue, collectionId, context, column, dispatch]);
+
+  useEffect(() => {
+    //Clear when the filter is cleared
+    // Use 'any' for empty value so the select stays controlled and shows "Any" in the dropdown
+    if (filter.value === undefined || filter.value === null) {
```

Comment:
You could probably check here if the filter value is *different* than filterRef.current.value instead of only handling the any/undefined (cleared) state, but make sure it doesn't cause an infinite loop 😅